File: flow-tests/test-frontend/addon-with-templates/src/main/java/org/vaadin/example/addon/AddonLitComponent.java
package org.vaadin.example.addon;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;

@Tag(AddonLitComponent.TAG)
@JsModule("./AddonLitComponent.ts")
public class AddonLitComponent extends LitTemplate {

    public static final String TAG = "addon-lit-component";

    @Id("label")
    private Span label;

    public void setLabel(String value) {
        label.setText(value);
    }
}

File: flow-tests/test-frontend/addon-with-templates/src/main/resources/META-INF/frontend/AddonLitComponent.ts
import { html, LitElement } from 'lit';

class AddonLitComponent extends LitElement {
  render() {
    return html`<div>
      <p>Add-on component</p>
      <span id="label">Default</span>
    </div>`;
  }
}

customElements.define('addon-lit-component', AddonLitComponent);
